### PR TITLE
UI Experiment

### PR DESCRIPTION
**Pull request by petehamilton**
This commit adds a shiny frontend. It's not a technical marvel but it
gets the job done!

Uses webpack for building front end assets and adds it into the build
pipeline/readme.

In terms of the UI, it now:

- Pulls out some high level stats:
  - Distance travelled
  - Est. distance to go
  - Est. total distance
  - Percentage complete
  - Next section of journey
  - Days on the road
- Shows a most recent update with image/text (changes on map click)
- Shows a fun fact - couple of comparisons to sizes of other things (mars bar calorie equivalent etc)

whereisgrey

Works on mobile too:

image

### CODE REVIEW

**Comment by petehamilton** [comment]
If I have time I'll do a follow up this evening to make the JS slightly more consistent.

File: app/static/kml.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// KmlLoader.load returns a promise which resolves to and array containing all
```
Comment (greysteil) [inline app/static/kml.js]: `an array`

**Comment by petehamilton** [comment]
@greysteil made a few changes. Responded re: caching with my reasoning but it's minor, happy to remove if you'd prefer!

**Comment by petehamilton** [comment]
I was also thinking it'd be cool to get more granularity on the "what's next" description. Right now it picks the next section, would be cool to show the next country. Working that out in browser (using a number of geocoding calls) is probably painfully expensive to offload onto everyone visiting so I might run it once and generate another KML which divides by country.

What do you think?

**Comment by petehamilton** [comment]
Also not 100% sure about splitting out the map from the info windows, another option was to have the stats + progress at the top, then the rest be a map from there down. 💭 ?

**Comment by petehamilton** [comment]
In fact, I like that idea better. Hang fire until I've put that in place...

**Comment by greysteil** [comment]
Having a new KML file that's split by country would be awesome - I'd be up for ditching the existing one in favour of that. Also agree with stats + progress at the top, then the rest being a map.

**Comment by greysteil** [comment]
I'll be on a plane this evening, and then not sure how much wifi I'll have, so I'm going to make you a collaborator on this and on Heroku so you can merge and push when done. Really love the new design!

**Comment by petehamilton** [comment]
@greysteil sounds good to me. Awesome catching up this weekend and hearing about your crazy journey so far - good luck with the next stretch!

File: app/app.rb [hunk 2]
```diff
@@ -35,15 +36,27 @@ def google_public_api_key
     Prius.get(:google_public_api_key)
   end
 
+  def days_on_the_road
+    ((latest_check_in.sent_at - first_check_in.sent_at) / 86400).round
+  end
+
+  def checkins
+    @checkins ||= CheckIn.order(sent_at: :asc).all
```
Comment (petehamilton) [inline app/app.rb]: 👍

**Comment by petehamilton** [comment]
Ok, reasonably happy with this now. Will make other changes in separate PRs.

Going to do a test deploy to Heroku to check the front end build pipeline works as expected then will merge this in.

**Comment by greysteil** [comment]
:shipit:

**Comment by petehamilton** [comment]
Finally got Heroku to play ball by adding the right buildpacks in the right order. Let's do it!

**Pull request merged**